Source/SequenceMethodCall.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Reflection;
+
+using Moq.Proxy;
+
+namespace Moq
+{
+	// this corresponds to a setup created by `mock.SetupSequence`
+	internal class SequenceMethodCall : MethodCall
+	{
+		// contains the responses set up with `CallBase`, `Pass`, `Returns`, and `Throws`
+		// TODO: the use of a queue without keeping track of the number of received calls, and tying each
+		// response to an expected invocation count, means that things won't work quite like they used to!
+		// this is possibly a breaking change right now.
+		private ConcurrentQueue<Tuple<ResponseKind, object>> responses;
```

Comment:
The mentioned todo should be taken seriously. With the new implementation, we can exhaust the set up sequence, then receive default values for a while, then configure some more responses, and get those back immediately until the sequence is exhausted again. This might have worked differently with the old implementation, because it kept track of the exact call count. At least add a unit test verifying this scenario before and after the change.